### PR TITLE
jQuery promise cleanup, convert some tests to async functions

### PR DESCRIPTION
**Pull request by martinpitt**
Some Friday evening doodling and cleanups. I at least found some bugs, too :grin: 

Most of the remaining jQuery promise API is hard to convert: They use `.stream()` or other Cockpit specific API, and that isn't compatible with Promise API. That should be fixed, but that requires actual thining :grin:

### CODE REVIEW

File: pkg/networkmanager/network-interface-members.jsx [hunk 1]
```diff
@@ -71,10 +71,7 @@ export const NetworkInterfaceMembers = ({
                         if (val) {
                             with_checkpoint(
                                 model,
-                                function () {
-                                    return member_con.activate(dev)
-                                            .fail(show_unexpected_error);
-                                },
+                                () => member_con.activate(dev).catch(show_unexpected_error),
```
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:74]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230721-193803-d973134a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface-members.jsx [hunk 2]
```diff
@@ -109,10 +103,7 @@ export const NetworkInterfaceMembers = ({
                                     onClick={syn_click(model, () => {
                                         with_checkpoint(
                                             model,
-                                            function () {
-                                                return (free_member_connection(member_con)
-                                                        .fail(show_unexpected_error));
-                                            },
+                                            () => free_member_connection(member_con).catch(show_unexpected_error),
```
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:106]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230721-193803-d973134a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface-members.jsx [hunk 3]
```diff
@@ -167,7 +158,7 @@ export const NetworkInterfaceMembers = ({
                             () => {
                                 return set_member(model, main_connection, main_connection.Settings,
                                                   cs.type, iface.Name, true)
-                                        .fail(show_unexpected_error);
+                                        .catch(show_unexpected_error);
```
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:161]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230721-193803-d973134a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface.jsx [hunk 4]
```diff
@@ -152,9 +152,7 @@ export const NetworkInterfacePage = ({
 
         function modify () {
             return dev.disconnect()
-                    .fail(function (error) {
-                        show_unexpected_error(error);
-                    });
+                    .catch(error => show_unexpected_error(error));
```
Comment (cockpituous) [inline pkg/networkmanager/network-interface.jsx:155]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230721-193803-d973134a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/packagekit/autoupdates.jsx [hunk 5]
```diff
@@ -241,7 +241,7 @@ const AutoUpdatesDialog = ({ backend }) => {
     function save(event) {
         setPending(true);
         backend.setConfig(enabled, type, day, time)
-                .always(Dialogs.close);
+                .finally(Dialogs.close);
```
Comment (cockpituous) [inline pkg/packagekit/autoupdates.jsx:244]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230721-193803-d973134a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/overview-cards/motdCard.jsx [hunk 6]
```diff
@@ -53,8 +53,8 @@ const MotdEditDialog = ({ text }) => {
                        <Button variant='primary'
                                onClick={() => cockpit.file("/etc/motd", { superuser: "try", err: "message" })
                                        .replace(value)
-                                       .done(Dialogs.close)
-                                       .fail(exc => {
+                                       .then(Dialogs.close)
+                                       .catch(exc => {
```
Comment (cockpituous) [inline pkg/systemd/overview-cards/motdCard.jsx:57]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230721-193803-d973134a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
Uh, the [tox test](https://github.com/cockpit-project/cockpit/actions/runs/5629119892/job/15253728177?pr=19120) apparently uses a dynamic ruff version:
```
not ok 9 /static-code/test-ruff
# test/common/testlib.py:968:18: RUF015 [*] Prefer `next(...)` over single element slice
# test/pytest/test_bridge.py:398:21: RUF015 [*] Prefer `next(...)` over single element slice
```
That doesn't seem too hard, I'll fix it along here. But if that gets in the way more often, we should remove ruff from tox runs.

File: pkg/lib/serverTime.js [hunk 7]
```diff
@@ -125,34 +125,17 @@ export function ServerTime() {
                     remote_offset = offsetms;
                     emit_changed();
                 })
-                .fail(function(ex) {
-                    console.log("Couldn't calculate server time offset: " + cockpit.message(ex));
-                });
+                .catch(ex => console.log("Couldn't calculate server time offset: " + cockpit.message(ex)));
```
Comment (cockpituous) [inline pkg/lib/serverTime.js:128]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface-members.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:74]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface-members.jsx [hunk 8]
```diff
@@ -83,10 +80,7 @@ export const NetworkInterfaceMembers = ({
                         } else if (dev) {
                             with_checkpoint(
                                 model,
-                                function () {
-                                    return dev.disconnect()
-                                            .fail(show_unexpected_error);
-                                },
+                                () => dev.disconnect().catch(show_unexpected_error),
```
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:83]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface-members.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:106]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface-members.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/networkmanager/network-interface-members.jsx:161]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/network-interface.jsx (hunk 4 above)
Comment (cockpituous) [inline pkg/networkmanager/network-interface.jsx:155]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/packagekit/autoupdates.jsx (hunk 5 above)
Comment (cockpituous) [inline pkg/packagekit/autoupdates.jsx:244]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/overview-cards/motdCard.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/systemd/overview-cards/motdCard.jsx:57]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19120-20230722-055749-01048432-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]
Nice cleanups, I can drop the cleanout jQuery promises from my todo now :+1:

**Pull request merged**